### PR TITLE
Premium Content Block: Update to useBlockProps

### PR DESCRIPTION
**Pull request by apeatling**
See https://github.com/Automattic/jetpack/issues/18864

#### Changes proposed in this Pull Request:

* Update the premium content block to use the block API version 2.

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Insert a Premium Content block
* Confirm that it still works as before in the editor and front end.
* Make sure that the deprecation works correctly coming from a previous verison of the block.

### CODE REVIEW

**Review by stacimc — DISMISSED** [review]
When I insert the block into the editor, I'm seeing block validation errors in the console: 
```Block validation: Expected attribute `class` of value `wp-block-premium-content-buttons wp-block-buttons`, saw `wp-block-buttons`.```

Screen Shot 2021-05-13 at 11 27 15 AM

**Comment by apeatling** [comment]
@glendaviesnz or @aaronrobertshaw Any idea why the deprecation I've added doesn't work in this scenario? Seems really simple and something that should work. What's missing?

File: projects/plugins/jetpack/extensions/blocks/premium-content/buttons/index.js [hunk 1]
```diff
@@ -29,6 +30,18 @@ const settings = {
 	edit,
 	save,
 	usesContext: [ 'premium-content/planId', 'premium-content/isPreview' ],
+	deprecated: [
+		{
+			attributes: {},
```
Comment (andrewserong) [inline projects/plugins/jetpack/extensions/blocks/premium-content/buttons/index.js:35]: I think to get this deprecation working, it also needs to specify the block supports that it's using (if I'm reading the [docs](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-deprecation/) correctly 🤔). Adding the following works for me:

```suggestion
			attributes: {},
			supports: {
				align: true,
				alignWide: false,
				lightBlockWrapper: true,
				inserter: false,
			},			
```

Then in the console, it looks like the markup is getting updated correctly:

image

**Review by andrewserong — COMMENTED** [review]
@apeatling I think it was just missing the `supports` key to get the deprecation working, I've added a suggested change 🙂

File: projects/plugins/jetpack/extensions/blocks/premium-content/buttons/index.js (hunk 1 above)
Comment (apeatling) [reply projects/plugins/jetpack/extensions/blocks/premium-content/buttons/index.js:35]: Thanks!

**Review by andrewserong — APPROVED** [review]
Thanks for fixing this up @apeatling! This is working nicely for me now. Existing (deprecated) block markup is successfully updated in the editor. Also confirmed that Subscribe patterns are rendering correctly in the pattern inserter (and on insertion, too) 👍

**Review by jeherve — APPROVED** [review]
This seems to test well for me. 👍

**Comment by jeherve** [comment]
This should be good to merge. I have removed the "Fixes" from the PR, since `__experimentalBlock` is also in use for the Video block.

**Comment by simison** [comment]
I noticed one more instance, was it left out intentionally from this PR?

https://github.com/Automattic/jetpack/blob/ba2dd6c30be137baad96aafb2466d0aed7e182e5/projects/plugins/jetpack/extensions/blocks/premium-content/login-button/edit.js#L12

**Comment by apeatling** [comment]
No just missed! Thanks for catching, for some reason my folder search didn't pick it up. I'll open a followup today.

**Comment by jeherve** [comment]
r226541-wpcom

**Pull request merged**